src/profiler.rs

```diff
@@ -142,6 +175,29 @@ pub struct SymbolizedAggregatedSample {
     pub count: u64,
 }
 
+impl fmt::Display for SymbolizedAggregatedSample {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let format_symbolized_stack = |symbolized_stack: &Vec<String>| -> String {
+            let mut scratch_string: String = String::new();
+            scratch_string.push_str("[\n");
+            for (i, symbol) in symbolized_stack.iter().enumerate() {
+                let cvtd = format!("{:3}: {},\n", i, symbol);
+                scratch_string.push_str(&cvtd);
+            }
+            scratch_string.push(']');
+            scratch_string
+        };
+        let ustack_rep = format_symbolized_stack(&self.ustack);
+        let kstack_rep = format_symbolized_stack(&self.kstack);
+        let final_rep = write!(
+            f,
+            "SymbolizedAggregatedSample:\npid: {}\nustack: {}\nkstack: {}\ncount: {}",
```

Comment:
This is lovely! Some small improvement we could make here is to use the formatting facilities from `fmt::Formatter`, something like:

```suggestion
             f.debug_struct("SymbolizedAggregatedSample")
                        .field("pid", &pid)
                        .field("ustack", &ustack)
                        .finish()
```